lwc/src/main/scala/quasar/physical/s3/S3LWFS.scala

```diff
@@ -26,12 +26,27 @@ import org.http4s.client.Client
 import org.http4s.Uri
 import scalaz.concurrent.Task
 
+// three functions comprise the LightweightFileSystem API
+// (and informally, the lightweight connector API itself)
+// since they're all implemented in their own files and
+// delegated to, I'll focus on the interface itself.
 final class S3LWFS(jsonParsing: S3JsonParsing, uri: Uri, client: Client) extends LightweightFileSystem {
 
+  // analogue of POSIX "ls" or Windows "dir", used to find
+  // the "immediate children" of a directory.
+  // on S3, object names roughly follow the standard
+  // folder1/folder2/file path format, which makes this
+  // function's implementation fairly natural.
   def children(dir: ADir): Task[Option[Set[PathSegment]]] = impl.children(client, uri, dir)
 
+  // reads from an S3 object containing JSON, which is parsed
+  // according to `jsonParsing`. streams results back.
   def read(file: AFile): Task[Option[Stream[Task, Data]]] = impl.read(jsonParsing, client, uri, file)
 
+  // does exactly what you'd guess, checks if a file (object)
```

Comment:
I'd remove "does exactly what you'd guess" since it's not very professional, imo. These docs are what other developers who want to write their own S3 connectors are going to read.